ql/src/java/org/apache/hadoop/hive/ql/io/orc/VectorizedOrcAcidRowBatchReader.java

```diff
@@ -885,7 +886,20 @@ static Path[] getDeleteDeltaDirsFromSplit(OrcSplit orcSplit,
     } else {
       throw new IllegalStateException("Split w/o base w/Acid 2.0??: " + path);
     }
-    return AcidUtils.deserializeDeleteDeltas(root, orcSplit.getDeltas(), pathToDeltaMetaData);
+
+    // filtering out delete_delta folders containing transactions that happened only before
+    // transactions of the split
+    List<AcidInputFormat.DeltaMetaData> deltas = orcSplit.getDeltas();
```

Comment:
wouldn't it be better if orcSplit.getDeltas() would already return only the relevant delete deltas?

Reply:
Yes, I agree it is better and I made the filtering or irrelevant deltas to happen on the split initialization as suggested.